src/Kentico.Xperience.Lucene.Core/Indexing/LuceneAutomaticReindexingBackgroundService.cs

```diff
@@ -0,0 +1,149 @@
+
+
+    private readonly TimeSpan assemblyVersionCheckInterval;
+
+
+    private const int MinimumCheckIntervalMinutes = 1;
+
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LuceneAutomaticReindexingBackgroundService"/> class.
+    /// </summary>
+    /// <remarks>This service is responsible for managing the automatic reindexing of Lucene indexes in the
+    /// background. It uses the provided dependencies to check the assembly version of indexes, determine if reindexing
+    /// is needed, and perform reindexing operations as necessary. The reindexing behavior is configured through the
+    /// provided <paramref name="luceneSearchOptions"/>.</remarks>
+    /// <param name="luceneAssemblyVersionInfoProvider">The <see cref="IInfoProvider{LuceneIndexAssemblyVersionItemInfo}"/> provides information about the assembly version of <see cref="LuceneIndex"/>s. Used to determine if reindexing is required.</param>
+    /// <param name="luceneClient">The <see cref="ILuceneClient"/> used to interact with the Lucene search infrastructure.</param>
+    /// <param name="luceneIndexManager">The <see cref="ILuceneIndexManager"/> which manages <see cref="LuceneIndex"/>s, including operations such as creating, updating, and deleting indexes.</param>
+    /// <param name="eventLogService">The <see cref="IEventLogService"/> logs events and errors related to the reindexing process.</param>
+    /// <param name="luceneSearchOptions"><see cref="IOptions{LuceneSearchOptions}"/> for Lucene search, including settings for post-startup reindexing.</param>
+    public LuceneAutomaticReindexingBackgroundService(
+        IInfoProvider<LuceneIndexAssemblyVersionItemInfo> luceneAssemblyVersionInfoProvider,
+        ILuceneClient luceneClient,
+        ILuceneIndexManager luceneIndexManager,
+        IEventLogService eventLogService,
+        IOptions<LuceneSearchOptions> luceneSearchOptions
+    )
+    {
+        this.luceneIndexManager = luceneIndexManager;
+        this.eventLogService = eventLogService;
+        this.luceneAssemblyVersionInfoProvider = luceneAssemblyVersionInfoProvider;
+        this.luceneClient = luceneClient;
+        reindexingOptions = luceneSearchOptions.Value.PostStartupReindexingOptions;
+        assemblyVersionCheckInterval = TimeSpan.FromMinutes(reindexingOptions?.CheckIntervalMinutes ?? 0);
+    }
+
+
+    /// <inheritdoc/>
+    protected override async Task ExecuteInternal(CancellationToken stoppingToken)
+    {
+        if (reindexingOptions is null || !reindexingOptions.Enabled || reindexingOptions.CheckIntervalMinutes < MinimumCheckIntervalMinutes)
+        {
+            return;
+        }
+
+        eventLogService.LogInformation(
+        "Kentico.Xperience.Lucene",
+        $"{nameof(LuceneAutomaticReindexingBackgroundService)}.{nameof(StartAsync)}",
+        "Lucene automatic reindexing if assembly version differs check started.");
+
+        using var timer = new PeriodicTimer(assemblyVersionCheckInterval);
```

Comment:
Based on the comment from PeriodicTimer constructor, the period argument must be larger than 1; otherwise, an ArgumentOutOfRangeException is thrown. On row assemblyVersionCheckInterval is set to 0 if reindexingOptions are null, so it will result in the exception. Is this intentional?